pkg/frontend/v2/frontend_scheduler_worker.go

```diff
@@ -374,14 +404,14 @@ func (w *frontendSchedulerWorker) enqueueRequest(loop schedulerpb.SchedulerForFr
 	durationTimer := prometheus.NewTimer(w.enqueueDuration)
 	defer durationTimer.ObserveDuration()
 
-	err := loop.Send(&schedulerpb.FrontendToScheduler{
-		Type:            schedulerpb.ENQUEUE,
-		QueryID:         req.queryID,
-		UserID:          req.userID,
-		HttpRequest:     req.request,
-		FrontendAddress: w.frontendAddr,
-		StatsEnabled:    req.statsEnabled,
-	})
+	frontendToSchedulerRequest, err := w.toSchedulerAdapter.frontendToSchedulerEnqueueRequest(req, w.frontendAddr)
+	if err != nil {
+		level.Warn(spanLogger).Log("msg", "received error while converting frontend request to scheduler request", "err", err)
```

Comment:
[nit] Probably wouldn't use the word "received" here, it might be misleading:

```suggestion
		level.Warn(spanLogger).Log("msg", "error while converting frontend request to scheduler request", "err", err)
```